src/main/java/kitchenpos/application/TableGroupService.java

```diff
@@ -1,89 +1,58 @@
+                             final TableGroupRepository tableGroupRepository) {
+        this.orderRepository = orderRepository;
+        this.orderTableRepository = orderTableRepository;
+        this.tableGroupRepository = tableGroupRepository;
     }
 
     @Transactional
-    public TableGroup create(final TableGroup tableGroup) {
-        final List<OrderTable> orderTables = tableGroup.getOrderTables();
-
-        if (CollectionUtils.isEmpty(orderTables) || orderTables.size() < 2) {
-            throw new IllegalArgumentException();
-        }
-
-        final List<Long> orderTableIds = orderTables.stream()
-                .map(OrderTable::getId)
-                .collect(Collectors.toList());
-
-        final List<OrderTable> savedOrderTables = orderTableDao.findAllByIdIn(orderTableIds);
-
-        if (orderTables.size() != savedOrderTables.size()) {
+    public TableGroupResponse create(final TableGroupCreateRequest request) {
+        final List<OrderTable> orderTables = orderTableRepository.findAllByIdIn(request.getOrderTables());
+        if (orderTables.size() != request.getOrderTables().size()) {
             throw new IllegalArgumentException();
         }
 
-        for (final OrderTable savedOrderTable : savedOrderTables) {
-            if (!savedOrderTable.isEmpty() || Objects.nonNull(savedOrderTable.getTableGroupId())) {
-                throw new IllegalArgumentException();
-            }
-        }
-
-        tableGroup.setCreatedDate(LocalDateTime.now());
-
-        final TableGroup savedTableGroup = tableGroupDao.save(tableGroup);
-
-        final Long tableGroupId = savedTableGroup.getId();
-        for (final OrderTable savedOrderTable : savedOrderTables) {
-            savedOrderTable.setTableGroupId(tableGroupId);
-            savedOrderTable.setEmpty(false);
-            orderTableDao.save(savedOrderTable);
-        }
-        savedTableGroup.setOrderTables(savedOrderTables);
+        final TableGroup tableGroup = new TableGroup(orderTables);
 
-        return savedTableGroup;
+        return TableGroupResponse.toResponse(tableGroupRepository.save(tableGroup));
     }
 
     @Transactional
     public void ungroup(final Long tableGroupId) {
-        final List<OrderTable> orderTables = orderTableDao.findAllByTableGroupId(tableGroupId);
-
+        final List<OrderTable> orderTables = orderTableRepository.findAllByTableGroupId(tableGroupId);
         final List<Long> orderTableIds = orderTables.stream()
                 .map(OrderTable::getId)
                 .collect(Collectors.toList());
 
-        if (orderDao.existsByOrderTableIdInAndOrderStatusIn(
-                orderTableIds, Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))) {
+        if (orderRepository.existsByOrderTableIdInAndOrderStatusIn(orderTableIds,
+                Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))) {
```

Comment:
여기서 아래와 같이 바꾸면 위 과정에서 수행한 id 추출과정을 생략할 수 있지 않을까요?

```suggestion
        if (orderRepository.existsByOrderTablesInAndOrderStatusIn(orderTables,
                Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL))) {
```

Reply:
오오 2단계에서 피드백 반영했으면 이렇게 고쳤을 것 같네요!

근데 지금 3단계에서 의존성 분리하고 보니 저렇게 되면 OrderRepository에서 OrderTable을 의존하게 될 것 같아서 Id로 확인하도록 하는 게 괜찮아 보이는데 주노는 어떻게 생각하시나요??

Reply:
간접참조를 하고있는 시점에서는 동의합니다~!